scipy/stats/_continuous_distns.py

```diff
@@ -3985,6 +3985,96 @@ def fit(self, data, *args, **kwds):
 laplace = laplace_gen(name='laplace')
 
 
+class laplace_asymmetric_gen(rv_continuous):
+    r"""An asymmetric Laplace continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `laplace_asymmetric` is
+
+    .. math::
+
+        f(x, \kappa) &= \frac{1}{\kappa+\kappa^{-1}}\exp(-x/\kappa),\quad x\ge0\\
+                     &= \frac{1}{\kappa+\kappa^{-1}}\exp(x\kappa),\quad x<0\\
+
+    for :math:`-\infty < x < \infty`, :math:`\kappa > 0`.
+
+    `laplace_asymmetric` takes ``kap`` as a shape parameter for
+    :math:`\kappa`. For :math:`\kappa = 1`, it is identical to a
+    Laplace distribution.
+
+    %(after_notes)s
+
+    See Also
+    --------
+    laplace : Laplace distribution
+
+    References
+    ----------
+    Kozubowski TJ and Podgórski K. A Multivariate and Asymmetric
+    Generalization of Laplace Distribution, Computational
+    Statistics 15, 531–540 (2000). https://doi.org/10.1007/PL00022717
+
+    %(example)s
+
+    """
+    def _rvs(self, kap):
+        return self._random_state.laplace_asymmetric(kap, self._size)
```

Comment:
self._random_state can be an instance of either RandomState or Generator from rumpy.random.

Reply:
Thanks. Would you please check (and correct with code if necessary) my educated guess on the right way to handle this?

https://github.com/scipy/scipy/blob/899ed457f538bc1acf4126c85d18a22de8663e94/scipy/stats/_continuous_distns.py#L4024-L4025

Reply:
Have this working now, thanks.